### ISSUE TITLE
Installing pdbpp on Python 3

### ISSUE THREAD

**Issue by antocuni** [pdbpp/pdbpp#46]
Originally reported by: **JB A (Bitbucket: [vardd](https://bitbucket.org/vardd), GitHub: Unknown)**

----------------------------------------

Hi, 

I am getting this error when trying to install pdbpp in my virtual_env in python 3.

> pip install pdbpp

Downloading/unpacking pdbpp
  Downloading pdbpp-0.8.2.tar.gz (40kB): 40kB downloaded
  Running setup.py (path:<...>py3/build/pdbpp/setup.py) egg_info for package pdbpp
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "<...>py3/build/pdbpp/setup.py", line 12, in <module>
        long_description = open(readme).read() + '\n\n' + open(changelog).read()
      File "<...>/lib/python3.4/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 293: ordinal not in range(128)
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "<...>py3/build/pdbpp/setup.py", line 12, in <module>

    long_description = open(readme).read() + '\n\n' + open(changelog).read()

  File "<...>py3/lib/python3.4/encodings/ascii.py", line 26, in decode

    return codecs.ascii_decode(input, self.errors)[0]

UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 293: ordinal not in range(128)

> pip list

docutils (0.12)
httplib2 (0.9)
Jinja2 (2.7.3)
MarkupSafe (0.23)
pip (1.5.6)
psutil (2.1.3)
py (1.4.26)
Pygments (2.0.1)
pytest (2.6.4)
PyYAML (3.11)
restview (2.1.1)
setuptools (2.1)
Sphinx (1.2.3)
sphinx-rtd-theme (0.1.6)
sphinxcontrib-napoleon (0.2.8)
                 

Thanks, 
Let me know if more info is needed

(Sorry for the formatting, I tried to fix it,  but apparently I cant...)

----------------------------------------
- Bitbucket: https://bitbucket.org/antocuni/pdb/issue/46

**Comment by antocuni** [comment]
*Original comment by* **Jim Fulton (Bitbucket: [j1m](https://bitbucket.org/j1m), GitHub: [j1m](https://github.com/j1m))**:

----------------------------------------

This hurts. Is there a work around?

**Comment by antocuni** [comment]
*Original comment by* **Jim Fulton (Bitbucket: [j1m](https://bitbucket.org/j1m), GitHub: [j1m](https://github.com/j1m))**:

----------------------------------------

Note that I'm on Ubuntu 16.04

**Comment by antocuni** [comment]
*Original comment by* **Jim Fulton (Bitbucket: [j1m](https://bitbucket.org/j1m), GitHub: [j1m](https://github.com/j1m))**:

----------------------------------------

I had the same problem on Ubuntu 14.04.

I finally got around this by hacking setup.py in my download cache to change the computation of long_description to:

    long_description = (
        open(readme, 'rb').read().decode('utf-8') + '\n\n' +
        open(changelog, 'rb').read().decode('utf-8')
        )

I couldn't see any obviously odd characters in the README or in the CHANGLOG. <shrug>

**Comment by antocuni** [comment]
*Original comment by* **Jim Fulton (Bitbucket: [j1m](https://bitbucket.org/j1m), GitHub: [j1m](https://github.com/j1m))**:

----------------------------------------

I should have mentioned that he root of the problem is that Python thinks my file-system encoding is ascii.

**Comment by antocuni** [comment]
*Original comment by* **Wouter Bolsterlee (Bitbucket: [wbolster](https://bitbucket.org/wbolster), GitHub: [wbolster](https://github.com/wbolster))**:

----------------------------------------

is there any chance this bug can be fixed?

```
import io
long_description = (
    io.open(readme, encoding='utf-8').read() + "\n\n" +
    io.open(changelog, encoding='utf-8').read())
```

should work for both py27 and py3

**Comment by antocuni** [comment]
*Original comment by* **Aleks Bunin (Bitbucket: [b-enlnt](https://bitbucket.org/b-enlnt), GitHub: Unknown)**:

----------------------------------------

I can't duplicate this on macOS, but can on Centos Linux... 

Anyway, pull request #48 should resolve this issue.

**Comment by antocuni** [comment]
*Original comment by* **Antonio Cuni (Bitbucket: [antocuni](https://bitbucket.org/antocuni), GitHub: [antocuni](https://github.com/antocuni))**:

----------------------------------------

Fix setup on Python 3.x.

Fixes #46.
Fixes #64.
